lib/ansible/modules/source_control/git.py

```diff
@@ -977,6 +1011,39 @@ def main():
     track_submodules = module.params['track_submodules']
 
     result.update(before=None)
+
+    if archive:
```

Comment:
This is behaving as expected now. :) I noticed one more thing. Since this module [supports check mode](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/source_control/git.py#L903) and archive will always change things (by creating directories, cloning, creating and archive - and replacing if it was pre-existing) lines 1016-1044 should only report the things that will change if check mode is used rather than actually implementing it. [Here](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/source_control/git.py#L1019-L1036) is an example of it being used elsewhere in this module.